src/foam/u2/view/BadgeView.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * @license
+ * Copyright 2021 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.u2.view',
+  name: 'BadgeView',
+  extends: 'foam.u2.View',
+
+  css: `
+    ^Wrapper{
+      font-family: /*%FONT1%*/ Roboto, 'Helvetica Neue', Helvetica, Arial, sans-serif;
+      font-size: 10px;
+      font-style: normal;
+      font-stretch: normal;
+      font-weight: 500;
+      height: 24px;
+      line-height: 24px;
+      letter-spacing: normal;
+      padding: 0 8px;
+      text-align: center;
+      width: 79px;
+    }
+   
+  `,
+
+  documentation:`Creates badges with rounded/squared sides based on display context`,
+
+  properties: [
+    {
+        class: 'Boolean',
+        name: 'isAttachedToCard'
+    },
+    {
+        class: 'String',
+        name: 'label'
+    },
+    {
+        class: 'String',
+        name: 'backgroundColor'
+    },
+    {
+        class: 'String',
+        name: 'color'
+    }
+  ],
+
+  methods: [
+    function initE() {
+      this
+        .addClass(this.myClass('Wrapper'))
+        .callIfElse(this.isAttachedToCard,
+          function() {
+            this.style({ 'border-radius': '0px 11.2px 11.2px 0px' });
```

Comment:
I don't think we should have the isAttachedToCard feature, but if we kept it, it would be easier to do this as:
this.style({'border-radius': this.isAttachedToCard ? '...' : '...'})

Reply:
And then it could be included into the .style() call below.